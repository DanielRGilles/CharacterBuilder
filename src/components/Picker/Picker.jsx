import React from 'react'

export default function Picker({head, middle, leg, setHead,  setLegs, setMiddle, setPhrase, handleClick}) {
    const headPics = ['fox', 'giraffee', 'eagle'];
    const middlePics = ['baby', 'pizza', 'wings'];
    const legPics = ['centaur', 'bones', 'fish'];
    return (
        <div className='picker-choose'>
            <label>
                Head
                <select value={head} onChange={(e) => setHead(e.target.value)}>
                    {headPics.map((pic) => (<option key={pic}>{pic}</option>
                    ))}
                </select>
            </label>
            <label>
                Middle
                <select value={middle} onChange={(e) => setMiddle(e.target.value)}>
                    {middlePics.map((pic) => (<option key={pic}>{pic}</option>
                    ))}
                </select>

            </label>
            <label>
                Legs
                <select value={leg} onChange={(e) => setLegs(e.target.value)}>
                    {legPics.map((pic) => (<option key={pic}>{pic}</option>
                    ))}
                </select>

            </label>
        </div>
    )
}
