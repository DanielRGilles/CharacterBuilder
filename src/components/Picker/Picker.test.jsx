import {screen, render} from '@testing-library/react';
import Picker from './Picker';

it('should render our Picker component', () => {
  //Render component you want to run the test on
  const {container} = render(<Picker />);

  const selectHead = screen.getByLabelText(/head/);
  const selectMiddle = screen.getByLabelText(/middle/);
  const selectLeg = screen.getByLabelText(/legs/);
  const input = screen.getByLabelText(/CatchPhrase/);

  expect(selectHead).toBeInTheDocument();
  expect(selectLeg).toBeInTheDocument();
  expect(selectMiddle).toBeInTheDocument();
  expect(input).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});