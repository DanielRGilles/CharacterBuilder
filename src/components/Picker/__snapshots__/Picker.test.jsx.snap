// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render our Picker component 1`] = `
<div>
  <div
    class="picker-choose"
  >
    <label>
      head
      <select>
        <option>
          fox
        </option>
        <option>
          giraffee
        </option>
        <option>
          eagle
        </option>
      </select>
    </label>
    <label>
      middle
      <select>
        <option>
          baby
        </option>
        <option>
          pizza
        </option>
        <option>
          wings
        </option>
      </select>
    </label>
    <label>
      legs
      <select>
        <option>
          centaur
        </option>
        <option>
          bones
        </option>
        <option>
          fish
        </option>
      </select>
    </label>
    <label>
      CatchPhrase
      <input
        placeholder="new catch phrase"
        type="text"
        value=""
      />
      <button>
        Add
      </button>
    </label>
  </div>
</div>
`;
